lib/datadog/tracing/contrib/rack/middlewares.rb

```diff
@@ -61,6 +61,8 @@ def call(env)
               Tracing.continue_trace!(trace_digest)
             end
 
+            return @app.call(env) if previous_request_span
```

Comment:
* 1️⃣ This change seems pretty big, since it applies to all users of rack, not just sinatra.

  It also seems somewhat deliberate that we didn't do this in the past -- the rest of the method has checks and does different things when there is a previous request span, and when there isn't.

  This may be a bit of a hard question to answer, but do you know why we didn't need this before?
  
* 2️⃣ It looks to me that the only coverage we have for this change is through the sinatra specs.

  That's... a bit weird to me, e.g. thinking that if we ever decide to remove sinatra (or move it to a separate library), that we lose coverage for rack. Can I convince you to add coverage for this change to the rack specs as well?

* 3️⃣ On a "cleaner code" note, I would suggest moving this `return` to be just after the assignment of `previous_request_span`, and to remove every other check for `previous_request_span.nil?` in this method, since we know that in the new version that will never be the case (except in the `ensure`, which should stay)

Reply:
1. What we currently have is technically correct, the Rack/Sinatra apps could be mounted on each other to create a layered architecture. We have received feedback such UI is confusing, hence trigger to make a change.

2. Totally make sense, more [specs](https://github.com/DataDog/dd-trace-rb/pull/2217/commits/19f5d7864a606fd3de84f0c532bec4f8fe6c20fa) added for rack instrumentation

Reply:
Looks good, thanks!